modules/standard/Time.chpl

```diff
@@ -610,6 +610,11 @@ enum day       { sunday=0, monday, tuesday, wednesday, thursday, friday, saturda
     var chpl_hour, chpl_minute, chpl_second, chpl_microsecond: int;
     var chpl_tz: shared Timezone?;
 
+    // Timezone awareness/naivety is part of the type so we can prohibit
+    // comparison of an aware vs naive time at compile-time.
+    @chpldoc.nodoc
+    param tz_aware : bool = false;
```

Comment:
Should the name of this be something more user-facing like `tzAware` (straw-person proposal)? It is nodoced here but it will show up elsewhere.